gui/builtinViews/fittingView.py

```diff
@@ -408,6 +408,9 @@ def swapItems(self, x, y, srcIdx):
             if mod1.slot != mod2.slot:
                 return
 
+            if not getattr(mod2, "modPosition"):
+                self.mods.modPosition = mod2.modPosition = dstRow
+
```

Comment:
Long story short there are times when we can have `modPosition` be empty.  This is just a check to see if it is, and if so we explicitly set it.  Looking at it, though, I think there's a better way of handling it.

It's a good example of what we have scattered throughout the code base, which is a larger issue.

```
if clone and mod2.isEmpty:
    sFit.cloneModule(self.mainFrame.getActiveFit(), srcIdx, mod2.modPosition)
```

We actually commit the same sin twice here, though one is worse than the other.  If `isEmpty` didn't exist, then we'd skip it, so that's not huge problem (though there's lots of times when we are checking for a value in the `if` and that would stacktrace).

The one that will stacktrace is `mod2.modPosition`.

We do this a _lot_ in the code base, we simply use variables assuming that they exist without any validation or defaults.

I'm going to rewrite this one to use `getattr` directly, I don't know why I didn't go that route in the first place.

Honestly these really should be rewritten to use `getattr` and a default value, but since that's littered _everywhere_ in the code I don't think we should tackle that directly.  Mostly just make a habit of _not_ accessing attributes directly, but using `getattr` and a default.